lib/src/metta/text.rs

```diff
@@ -61,147 +64,395 @@ impl Tokenizer {
 
 }
 
+/// The meaning of a parsed syntactic element, generated from a substring in the input text
+#[derive(Clone, Debug)]
+pub enum SyntaxNodeType {
```

Comment:
This is a design point I actually wrestled with for several days, and I tried it a few different ways and I felt there were efficiency and code clarity tradeoffs to each approach.  But with the benefit of a couple days away, I think you might be right this can be improved.  Let me spend a few more minutes on it.

Reply:
Ok.  I feel like the code is cleaner with the changes in faffafb.

It's nice because it's now possible to perform the parse without access to the tokenizer, or creating any atoms at all.  But I suspect this win is temporary because we'll want to actually make the atoms in order to get their types, when we introduce MeTTa-type-dependent styling.